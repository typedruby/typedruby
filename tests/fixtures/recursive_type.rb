# Test that unification won't generate recursive types
def recursive => nil
  a = []
  a << a
  a << 1
  nil
end
