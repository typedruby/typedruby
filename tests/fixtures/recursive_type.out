error: Could not match types:

        @ __ROOT__/tests/fixtures/recursive_type.rb:5
      5 |    a = []
                 ^^ t1, with:
        @ __ROOT__/tests/fixtures/recursive_type.rb:5
      5 |    a = []
                 ^^ Array::[t1]

        - arising from an attempt to match:

        @ (builtin stdlib):790
    790 |    def <<(ElementType item) => :self; end
                    ^^^^^^^^^^^ t1, with:
        @ __ROOT__/tests/fixtures/recursive_type.rb:6
      6 |    a << a
                  ^ Array::[t1]
        @ __ROOT__/tests/fixtures/recursive_type.rb:6
      6 |    a << a
             ^^^^^^ in this expression
